Dockerfile

```diff
@@ -0,0 +1,63 @@
+FROM ubuntu:14.04
+MAINTAINER Cezary Baginski <cezary@chronomantic.net>
+
+# Packages needed to install RVM and run Bundler gem commands
+RUN apt-get update -qq && apt-get -y install ca-certificates curl git-core --no-install-recommends && rm -rf /var/lib/apt/lists/* /var/cache/apt/archives/*.deb /var/cache/apt/archives/partial/*.deb /var/cache/apt/*.bin
+
+# Create guest user early (before rvm) so uid:gid are 1000:000
+RUN useradd -m -s /bin/bash guest
+
+# Temporarily install RVM as root - just for requirements
+RUN gpg --keyserver hkp://keys.gnupg.net --recv-keys 409B6B1796C275462A1703113804BB82D39DC0E3
+RUN curl -L get.rvm.io | bash -s stable
+RUN bash -l -c 'rvm requirements 2.2.1'
+RUN bash -l -c 'echo yes | rvm implode'
+
+# Fix locale
+ENV DEBIAN_FRONTEND noninteractive
+RUN dpkg-reconfigure locales && locale-gen en_US.UTF-8 && /usr/sbin/update-locale LANG=en_US.UTF-8
+RUN echo 'en_US.UTF-8 UTF-8' >> /etc/locale.gen && locale-gen
+ENV LC_ALL C.UTF-8
+ENV LANG C.UTF-8
+ENV LANGUAGE C.UTF-8
+
+USER guest
+ENV HOME /home/guest
+WORKDIR /home/guest
+
+# Install RVM as guest
+RUN gpg --keyserver hkp://keys.gnupg.net --recv-keys 409B6B1796C275462A1703113804BB82D39DC0E3
+RUN /bin/bash -l -c "echo 'gem: --no-ri --no-rdoc' > ~/.gemrc"
+RUN curl -L get.rvm.io | bash -s stable
+RUN /bin/bash -l -c "rvm install 2.3.0"
+RUN /bin/bash -l -c "gem install bundler --no-ri --no-rdoc"
+
+ONBUILD ENV USER guest
+ONBUILD WORKDIR /home/guest
+ONBUILD RUN /bin/bash -l -c "source /home/$USER/.rvm/scripts/rvm"
+
+# Aruba specifics start here
+
+USER root
+
+# Zsh
+RUN apt-get update -qq && apt-get -y install zsh --no-install-recommends && rm -rf /var/lib/apt/lists/* /var/cache/apt/archives/*.deb /var/cache/apt/archives/partial/*.deb /var/cache/apt/*.bin
+
+# Python
+RUN apt-get update -qq && apt-get -y install python --no-install-recommends && rm -rf /var/lib/apt/lists/* /var/cache/apt/archives/*.deb /var/cache/apt/archives/partial/*.deb /var/cache/apt/*.bin
+
+# Java (for javac)
+RUN apt-get update -qq && apt-get -y install openjdk-7-jdk --no-install-recommends && rm -rf /var/lib/apt/lists/* /var/cache/apt/archives/*.deb /var/cache/apt/archives/partial/*.deb /var/cache/apt/*.bin
+
+# Add gemrc for possible gem mirror config
+ADD docker.gemrc /home/guest/.gemrc
+RUN chown -R guest:guest /home/guest/.gemrc
+
+# Cache needed gems - for faster test reruns
+ADD Gemfile Gemfile.lock aruba.gemspec /home/guest/cache/aruba/
```

Comment:
I'm not sure why. It's not a web app we're using. And I don't think anyone contributing to aruba is using /srv/app for working on Ruby projects. They usually work in folders like /home/user/work.

I'm actually trying to avoid the "server mentality" here, because the priorities are opposite for "development". Busybox is great for servers but useless for getting work done. 

I don't want people to think this is about "building your own Travis". It's just "test isolation", pretty much.